### ISSUE TITLE
[MaterialDatePicker] Select new date when a year is changed

### ISSUE THREAD

**Issue by CoolMind** [material-components/material-components-android#1723]
When we change a year in a `DatePicker`, it scroll to another date, but it is not selected. When we later press "OK", the date doesn't change (strange that even "OK" is enabled). The user has to click a new date and then press "OK". So, he has to make a unneeded click.

I suppose, a new date (the same day and month, except 29 Febuary, and a new year) should be selected. If the user then presses "OK", it should be assigned. If he reconsidered and presses "Cancel", the date won't change.

![image](https://user-images.githubusercontent.com/10630436/93085246-f6416280-f69d-11ea-985e-38e9b97b3566.png)
![image](https://user-images.githubusercontent.com/10630436/93085298-0eb17d00-f69e-11ea-9cba-a8c85f6facc6.png)
![image](https://user-images.githubusercontent.com/10630436/93085091-b8443e80-f69d-11ea-9a89-665e0b14381b.png)

**Comment by iPaulPro** [comment]
Yeah, this is a really poor user experience, and causing a lot of issues for users. If a user selects a month and date, **then** changes the year, it keeps the month but loses the day selection. If the user presses "OK" at this point, it submits the first date that was selected.

I'd argue that this is a bug, and not a feature request.

**Comment by CoolMind** [comment]
@iPaulPro thanks, agree with you!
